core/src/main/java/com/linecorp/armeria/server/decoding/DecodingService.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.decoding;
+
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+
+import java.util.Map;
+import java.util.function.Function;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Streams;
+
+import com.linecorp.armeria.client.encoding.StreamDecoderFactory;
```

Comment:
Oops... this is `client` module. 😱 🤔
It is not natural to use client-side code on server-side.

Reply:
I also think `StreamDecoderFactory` class move from client to server...but I don't do this because i'm not a main contributer